CMakeLists.txt

```diff
@@ -0,0 +1,168 @@
+cmake_minimum_required(VERSION 3.4)
+
+set(CMAKE_MODULE_PATH "${CMAKE_SOURCE_DIR}/tools/cmake" ${CMAKE_MODULE_PATH})
+
+project(realm VERSION 2.3.0)
```

Comment:
I think we ought to read the `git-describe` output and pass it to `project()` here, and use the built-in `PROJECT_VERSION` variable elsewhere instead of `REALM_VERSION`.